hyperspy/_signals/eds.py

```diff
@@ -644,6 +644,10 @@ def get_lines_intensity(self,
                                                            FWHM_MnKa='auto')
             element, line = utils_eds._get_element_and_line(Xray_line)
             img = self.isig[window[0]:window[1]].integrate1D(-1)
+            if np.issubdtype(img.data, np.integer):
```

Comment:
This command requires two dtypes. It does not fail because the test does not actually reach this part of the code (i.e. tests something else)